symbolic-base/src/ZkFold/Base/Protocol/Plonkup/PlonkConstraint.hs

```diff
@@ -53,19 +54,19 @@ instance (Ord a, Arbitrary a, Binary a, KnownNat i) => Arbitrary (PlonkConstrain
 
 toPlonkConstraint :: forall a i . (Ord a, FiniteField a, KnownNat i) => Poly a (Var a (Vector i)) Natural -> PlonkConstraint i a
 toPlonkConstraint p =
-    let xs    = toList (variables p)
+    let xs    = Just <$> toList (variables p)
         perms = nubOrd $ map (take 3) $ permutations $ case length xs of
-            0 -> [ConstVar one, ConstVar one, ConstVar one]
```

Comment:
Is this change needed?

Reply:
I do not see how else we can separate `ConstVar one` here as a placeholder for an impossible variable and `ConstVar one` as an input polynomial. `perms` will hold exactly the same value if the input polynomial does not have any variables and if it is `ConsyVar one`

Reply:
But do we need to separate a placeholder from an input polynomial?